x/incentives/keeper/distribute.go

```diff
@@ -308,29 +320,105 @@ func (k Keeper) AllocateAcrossGauges(ctx sdk.Context) error {
 	return nil
 }
 
-// calcSplitPolicyCoins calculates tokens to split given a policy and groupGauge.
```

Comment:
Note to reviewer: the rest of this section below this line would probably be best to review in split view. The entire function was re-implemented, so the line-by-line changes are not relevant, and it seems the diff got messed up when merging with main